front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/handleParseFiles.ts

```diff
@@ -9,71 +9,94 @@ export const handleFileReadingError = (error: Error) => {
   console.error('File reading error:', error);
 };
 
-export const handleJsonParsingError = (error: Error, dispatch: Dispatch) => {
-  console.error('Error parsing JSON:', error);
-  dispatch(
-    setFailure({
-      name: t('errorMessages.error'),
-      message: t('errorMessages.errorInvalidJSONFormat'),
-    })
-  );
-};
+const validateTrainSchedules = (
+  importedTrainSchedules: Record<string, unknown>[]
+): TrainScheduleBase[] | null => {
+  const isInvalidTrainSchedules = importedTrainSchedules.some((trainSchedule) => {
+    if (
+      ['constraint_distribution', 'path', 'rolling_stock_name', 'start_time', 'train_name'].some(
+        (key) => !(key in trainSchedule)
+      ) ||
+      !Array.isArray(trainSchedule.path)
+    ) {
+      return true;
+    }
+    const hasInvalidSteps = trainSchedule.path.some((step) => ['id'].some((key) => !(key in step)));
```

Comment:
Can't we do that instead ? (since we check only the `id` key)
```suggestion
    const hasInvalidSteps = trainSchedule.path.some((step) => !('id' in step));
```

Reply:
Good catch, thanks !